connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockActionTranslator.java

```diff
@@ -106,9 +108,27 @@ public void translate(PlayerActionPacket packet, GeyserSession session) {
                 // Handled in BedrockInventoryTransactionTranslator
                 break;
             case START_BREAK:
-                ClientPlayerActionPacket startBreakingPacket = new ClientPlayerActionPacket(PlayerAction.START_DIGGING, new Position(packet.getBlockPosition().getX(),
-                        packet.getBlockPosition().getY(), packet.getBlockPosition().getZ()), BlockFace.values()[packet.getFace()]);
-                session.sendDownstreamPacket(startBreakingPacket);
+                Matcher m = null;
+                if(session.getConnector().getConfig().isCacheChunks()) {
+                    //check if Chunk Cache is enabled
+                    int fireLocation = session.getConnector().getWorldManager().getBlockAt(session, Vector3i.from(packet.getBlockPosition().getX(),
+                            packet.getBlockPosition().getY() + 1, packet.getBlockPosition().getZ()));
+                    String fireBlock = BlockTranslator.getJavaIdBlockMap().inverse().get(fireLocation);
+                    Pattern r = Pattern.compile("minecraft:fire");
+                    m = r.matcher(fireBlock);
+                }
+                //Check if fire is present to destroy
+                if(m.find() && packet.getFace() == 1) {
```

Comment:
This will throw a NullPointerException if chunk caching is disabled because the matcher is never set. Though I mentioned that regex is unecessary in a previous comment, may be worth keeping this in mind just incase 🙂